zerver/tests/test_message_summary.py

```diff
@@ -0,0 +1,103 @@
+import os
+import warnings
+from unittest import mock
+
+import orjson
+from django.conf import settings
+from typing_extensions import override
+
+from analytics.models import UserCount
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.views.message_summary import INPUT_COST_PER_GIGATOKEN, OUTPUT_COST_PER_GIGATOKEN
+
+warnings.filterwarnings("ignore", category=UserWarning, module="pydantic")
+warnings.filterwarnings("ignore", category=DeprecationWarning, module="pydantic")
+warnings.filterwarnings("ignore", category=DeprecationWarning, module="litellm")
+# Avoid network query to fetch the model cost map.
+os.environ["LITELLM_LOCAL_MODEL_COST_MAP"] = "True"
+import litellm
+
+# Fixture file to store recorded responses
+LLM_FIXTURES_FILE = "zerver/tests/fixtures/litellm/summary.json"
+
+
+class MessagesSummaryTestCase(ZulipTestCase):
+    @override
+    def setUp(self) -> None:
+        super().setUp()
+        self.user = self.example_user("iago")
+        self.topic_name = "New feature launch"
+        self.channel_name = "Zulip features"
+
+        self.login_user(self.user)
+        self.subscribe(self.user, self.channel_name)
+        content = "Zulip just launched a feature to generate summary of messages."
+        self.send_stream_message(
+            self.user, self.channel_name, content=content, topic_name=self.topic_name
+        )
+
+        content = "Sounds awesome! This will greatly help me when catching up."
+        self.send_stream_message(
+            self.user, self.channel_name, content=content, topic_name=self.topic_name
+        )
```

Comment:
Generally as a pattern, one doesn't want to send messages in a `setUp` function, because it's invisible when reading the code of individual tests, and folks will copy/paste the test to write a new one not realizing it'll do these steps.

Probably fine for now.